lang/wasm.cc

```diff
@@ -115,10 +115,31 @@ struct init_arg_visitor {
         if (serialized_size > std::numeric_limits<int32_t>::max()) {
             throw wasm::exception(format("Serialized parameter is too large: {} > {}", serialized_size, std::numeric_limits<int32_t>::max()));
         }
-        auto grown = memory.grow(store, 1 + (sizeof(int32_t) + serialized_size - 1) / WASM_PAGE_SIZE); // for fitting serialized size + the buffer itself
-        if (!grown) {
-            throw wasm::exception(format("Failed to grow wasm memory to {}: {}", serialized_size, grown.err().message()));
+        auto malloc_export = instance.get(store, "_scylla_malloc");
```

Comment:
Before doing this, we should check `_scylla_wasm_abi` to see which style of parameter passing we should use. I'm fine with deferring that to a later patch.

This patch however should update the ABI documentation.

I also want a runnable example/test.

Reply:
I've added the `_scylla_abi` symbol and the documentation patch, all tests in `cql-pytest/test_wasm.py` do use the new symbol, but a more interesting example will need to wait until after adding the Rust dependency